src/main/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidator.java

```diff
@@ -0,0 +1,286 @@
+            addError(errors, incorrectTotal, getJsonPath(subResource, COST_AT_PERIOD_END));
+        }
+
+    }
+
+    private Long getTransfers(IntangibleAssetsResource intangibleAssetsResource) {
+
+        return Optional.ofNullable(intangibleAssetsResource)
+                .map(IntangibleAssetsResource::getCost)
+                .map(Cost::getTransfers)
+                .orElse(0L);
+    }
+
+    private Long getRevaluations(IntangibleAssetsResource intangibleAssetsResource) {
+
+        return Optional.ofNullable(intangibleAssetsResource)
+                .map(IntangibleAssetsResource::getCost)
+                .map(Cost::getRevaluations)
+                .orElse(0L);
+    }
+
+    private Long getDisposals(IntangibleAssetsResource intangibleAssetsResource) {
+
+        return Optional.ofNullable(intangibleAssetsResource)
+                .map(IntangibleAssetsResource::getCost)
+                .map(Cost::getDisposals)
+                .orElse(0L);
+
+    }
+
+    private Long getAdditions(IntangibleAssetsResource intangibleAssetsResource) {
+
+        return Optional.ofNullable(intangibleAssetsResource)
+                .map(IntangibleAssetsResource::getCost)
+                .map(Cost::getAdditions)
+                .orElse(0L);
+    }
+
+    private Long getCostAtPeriodStart(IntangibleAssetsResource intangibleAssetsResource) {
+        return Optional.ofNullable(intangibleAssetsResource)
+                .map(IntangibleAssetsResource::getCost)
+                .map(Cost::getAtPeriodStart)
+                .orElse(0L);
+    }
+
+    private void validatePresenceOfSingleYearFilerFields(IntangibleAssetsResource intangibleAssetsResource, Errors errors, IntangibleSubResource intangibleSubResource, List<IntangibleSubResource> invalidSubResource) {
+
+        boolean subResourceInvalid = false;
+        if(intangibleAssetsResource.getCost() != null && intangibleAssetsResource.getCost().getAtPeriodStart() != null) {
+            addError(errors, unexpectedData, getJsonPath (intangibleSubResource, COST_AT_PERIOD_START));
+            subResourceInvalid = true;
+        }
+
+        if(intangibleAssetsResource.getNetBookValueAtEndOfCurrentPeriod() != null) {
+
+            subResourceInvalid = isCostSingleYearSubResourceInvalid(intangibleSubResource, intangibleAssetsResource.getCost(),  errors, subResourceInvalid);
+
+        } else {
+            if(hasSingleYearFilerNonNetBookValueFieldsSet(intangibleAssetsResource)) {
+                addError(errors, valueRequired, getJsonPath(intangibleSubResource, NET_BOOK_VALUE_CURRENT_PERIOD ));
+                subResourceInvalid = true;
+            }
+        }
+
+        if(subResourceInvalid) {
+            invalidSubResource.add(intangibleSubResource);
+        }
+
+    }
+
+    private boolean hasSingleYearFilerNonNetBookValueFieldsSet(IntangibleAssetsResource intangibleAssetsResource) {
+
+        if (intangibleAssetsResource.getCost() != null &&
```

Comment:
Rather than an if / else, you can just return the conditional

Reply:
Amended the code as suggested 👍